src/debug.rs

```diff
@@ -14,11 +14,11 @@
 
 // Generates an implementation of the Debug trait for a type that defers to the
 // Debug implementation for a given field.
-macro_rules! derive_debug_from_field {
-    ($typename:ty, $fieldname:ident) => {
+macro_rules! derive_debug_via_self {
+    ($typename:ty, self.$($tt:tt)+) => {
```

Comment:
Interesting, I didn't realize we could enforce that the expression starts with `self` like this.